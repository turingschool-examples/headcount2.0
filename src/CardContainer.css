.cardContainer-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(255px, 1fr));
  grid-gap: 15px;
  padding: 10px;
  margin: 10px;
  border-radius: 25px;
  border: cornflowerblue;
  align-items: center;
}

.cardContainerHeading {
  color: cornflowerblue; 
  text-align: center;
}