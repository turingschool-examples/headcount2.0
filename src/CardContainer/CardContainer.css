.card-container {
  display: flexbox;
  flex-wrap: wrap;
  border: black 1px solid;
  display: flex;
  flex-wrap: wrap;
  width: 84%;
  height: 500px;
  margin: 5px auto;
  justify-content: center;
  overflow-y: scroll;
}