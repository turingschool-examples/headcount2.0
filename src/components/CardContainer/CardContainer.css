/*CARDCONTAINER COMPONENT RELATED CSS GOES HERE*/
.CardContainer {
  grid-row: 3 / 4;
  grid-column: 1 / 2;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 15px;
}