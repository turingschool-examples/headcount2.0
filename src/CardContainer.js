import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './CardContainer.css';


const CardContainer = ( {displayData, dataSet} ) => {

  
  const cards = displayData.map(district => {
    return <Card 
            key={`card ${district.location}`}
            {...district} />
  })

  return(
    <div>
      <h3 className='cardContainerHeading'>{dataSet} data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.</h3>
      <section className='cardContainer-grid'>
        { cards }
      </section>
    </div>
  )
}

CardContainer.propTypes = {
  displayData: PropTypes.array.isRequired
}

export default CardContainer;