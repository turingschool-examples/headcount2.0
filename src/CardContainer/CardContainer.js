import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';
import './CardContainer.css'

const CardContainer = ({districtsData}) => {
  // console.log(districtsData);
  
  const districtCards = Object.keys(districtsData).map((key, index) => {
    return (
      <Card key={index} district={districtsData[key]} />
    )
  });

  return(
    <div className="card-container">
      {districtCards}
    </div>
  )
}

CardContainer.propTypes = {
  districtsData: PropTypes.any.isRequired
}

export default CardContainer;

      