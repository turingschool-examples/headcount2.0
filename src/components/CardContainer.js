import React, { Component } from 'react';
import Card from './Card';
import '../styles/CardContainer.css';


const CardContainer = (props) => {

  const dataArray = Object.keys(props.allDistrictData)
  const newCard = dataArray.map((location, index) => <Card data={ props.allDistrictData[location]} key={index} compareCards={props.compareCards}/>)
  
  return (
    <section className='container-wrap'>
      <h2>School Data:</h2>
      { newCard }
    </section>
  )
}

export default CardContainer