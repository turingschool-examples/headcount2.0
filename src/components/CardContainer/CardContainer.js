import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

import './CardContainer.css';

const CardContainer = props => {
  const { districts } = props;

  const districtData = districts.map(district => <Card {...district} key={district.location} />);

  return (
    <div>
      <h1>Card Container</h1>
      <div className="card-container">{districtData}</div>
    </div>
  );
};

CardContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CardContainer;
