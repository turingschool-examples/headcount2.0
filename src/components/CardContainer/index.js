import React from 'react'
import PropTypes from 'prop-types'
import Card from '../Card'
import './CardContainer.css'

const CardContainer = ({districts, adjustComparisons}) => {
  const cards = Object.keys(districts).map(district => {
    const stats = districts[district].stats

    return <Card key={`${district}-district`}
                 location={district}
                 stats={stats}
                 adjustComparisons={adjustComparisons}
           />
  })

  return(
    <main className="CardContainer">
       { cards }
    </main>
  )
}

CardContainer.propTypes = {
  districts: PropTypes.object.isRequired
}

export default CardContainer