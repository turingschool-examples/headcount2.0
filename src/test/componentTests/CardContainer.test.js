import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from '../../components/CardContainer';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<CardContainer/>)
  expect(renderedComponent).toMatchSnapshot()
})