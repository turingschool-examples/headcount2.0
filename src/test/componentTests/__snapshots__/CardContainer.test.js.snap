// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Cards area
</h2>,
        undefined,
      ],
      "className": "container-wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cards area",
        },
        "ref": null,
        "rendered": "Cards area",
        "type": "h2",
      },
      undefined,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Cards area
</h2>,
          undefined,
        ],
        "className": "container-wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cards area",
          },
          "ref": null,
          "rendered": "Cards area",
          "type": "h2",
        },
        undefined,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
